ct-app/economic_handler/economic_handler.py

```diff
@@ -539,6 +546,28 @@ def block_rpch_nodes(
         }
         return "dict_excluding_rpch_nodes", merged_metrics_subgraph_topology
 
+    def block_ct_nodes(self, merged_metrics_dict: dict):
+        """
+        Exludes nodes from the ct distribution that are connected to
+        Netwatcher/Postman modules.
+        :param: merged_metrics_dict (dict): merged topology, subgraph, database data
+        :returns: (dict): dictionary excluding ct-app instances
+        """
+        excluded_nodes = set()  # update a set to avoid duplicates
+
+        # Collect all unique node_addresses from the input data
+        for data in merged_metrics_dict.values():
+            excluded_nodes.update(data["node_addresses"])
+
+        # New dictionary excluding entries with keys in the exclusion set
+        metrics_dict_excluding_ct_nodes = {
+            key: value
+            for key, value in merged_metrics_dict.items()
+            if key not in excluded_nodes
+        }
+
+        return "dict_excluding_ct_nodes", metrics_dict_excluding_ct_nodes
+
```

Comment:
I understand the logic for now. Once everything will be splited in some nice asynchronous tasks, this method should be a standalone one, that do not rely on the `merged_metrics_subgraph_tolopology`.
The easy solution I see is to query the database, access the content of the column "node_address" and get the unique values of this column. This should contain the same information as the one you are gathering with your method.